astropy/visualization/wcsaxes/tests/test_display_world_coordinates.py

```diff
@@ -117,3 +123,18 @@ def test_cube_coords(self, tmpdir):
         fig.canvas.key_press_event(event1.key, guiEvent=event1)
         string_pixel = ax._display_world_coords(0.523412, 0.523412)
         assert string_pixel == "0.523412 0.523412 (pixel)"
+
+    @pytest.mark.remote_data(source='astropy')
+    def test_plot_coord_3d_transform(self):
+        filename = get_pkg_data_filename('galactic_center/gc_msx_e.fits')
+        wcs = WCS(filename)
+        data = fits.getdata(filename)
+
+        coord = SkyCoord(0 * u.kpc, 0 * u.kpc, 0 * u.kpc, frame='galactocentric')
+
+        fig = plt.figure()
+        ax = fig.add_subplot(1, 1, 1, projection=wcs)
+        ax.imshow(data)
+        point, = ax.plot_coord(coord, 'ro')
+
+        np.testing.assert_allclose(point.get_xydata()[0], [0, 0], atol=1e-4)
```

Comment:
I'm surprised that this would be at [0,0], it should be near the center of the image.

Reply:
yeah this one confused me as well. is `get_xydata` in pixels?

Reply:
Looks like it should be 😕

Reply:
Ah no I think it might be the data before applying the ``transform``. So to some extent this test is correct but I'd still be happier if we can figure out how to get the transformed data.

Reply:
You might need ``point.get_transform().transform(point.get_xydata())``

Reply:
That didn't work either, I can't see any other sensible solution other than this here :confused: